### PR TITLE
updates algorithm docs

### PR DESCRIPTION
**Pull request by melonakos**
Edited the algorithm.(dox/h) Doxygen documentation prose. This does not include new examples.

Description
-----------
Better prose and function prototypes.

Changes to Users
----------------
None

Checklist
---------
- [x] Rebased on the latest master
- [x] Code compiles
- [x] Functions documented

### CODE REVIEW

File: include/af/algorithm.h [hunk 1]
```diff
@@ -1095,246 +1139,235 @@ extern "C" {
...
-       C Interface for counting total number of non-zero values in an array,
-       while returning an af_array
+       C Interface to count non-zero values over all dimensions.
 
-       \param[out] out contain the number of non-zero values in \p in.
-       \param[in] in is the input array
-       \return \ref AF_SUCCESS if the execution completes properly
+       \param[out] out count
+       \param[in]  in  input array
+       \return         \ref af_err
```
Comment (syurkevi) [inline include/af/algorithm.h:1263]: Let's tweak everywhere. ex:
af_err, will be AF_SUCCESS if function executes properly

**Review by syurkevi — CHANGES_REQUESTED** [review]

File: docs/details/algorithm.dox [hunk 2]
```diff
@@ -105,216 +112,218 @@ s16                 | s32
...
+The keys' input type must be integer (s32 or u32).
 
-The input keys must be a 1-D vector matching the size of the reduced dimension.
-In the case of multiple dimensions in the input values array, the dim parameter
-specifies which dimension to reduce along.  An example of multi-dimensional
-reduce by key can be seen below:
+The output type is `b8`.
 
-\snippet test/reduce.cpp ex_reduce_alltrue_by_key_dim
+The keys array must be 1-dimenstional matching the size of the reduced
```
[the remark this answers is not in the document]
Comment (melonakos) [reply docs/details/algorithm.dox:229]: fixed

File: include/af/algorithm.h (hunk 1 above)
Comment (melonakos) [reply include/af/algorithm.h:1263]: fixed

**Comment by syurkevi** [comment]
superseded by #3485

**Pull request closed without merging**